stream_reader.go

```diff
@@ -32,41 +32,43 @@ func (stream *streamReader[T]) Recv() (response T, err error) {
 
 	var emptyMessagesCount uint
 
-waitForData:
-	line, err := stream.reader.ReadBytes('\n')
-	if err != nil {
-		respErr := stream.errAccumulator.unmarshalError()
-		if respErr != nil {
-			err = fmt.Errorf("error, %w", respErr.Error)
+	for {
```

Comment:
Due to added level of indentaion I wonder if we should factor out `for` body in the separate function (or maybe parts of the `for` body)

Reply:
I agree with you. Extracting the big, deeply nested for-loop into a separate method would be a good idea.

I will do so👍